crates/rust-analyzer/src/global_state.rs

```diff
@@ -298,14 +330,26 @@ impl GlobalState {
                 })
                 .map(|(file_id, (_, change))| vfs::ChangedFile { file_id, ..change })
                 .collect();
-
             let mut workspace_structure_change = None;
             // A file was added or deleted
             let mut has_structure_changes = false;
             let mut bytes = vec![];
             let mut modified_rust_files = vec![];
+
             for file in changed_files {
                 let vfs_path = vfs.file_path(file.file_id);
+                if let Some(("rust-analyzer", Some("toml"))) = vfs_path.name_and_extension() {
+                    // Remember ids to use them after `apply_changes`
+                    modified_ratoml_files.insert(file.file_id, vfs_path.clone());
+                    if vfs_path == self.config.user_config_path() {
+                        user_config_file_id = Some(file.file_id);
+                    }
+
+                    if vfs_path == &root_vfs_path {
+                        root_path_ratoml = Some(file.file_id);
+                    }
```

Comment:
We can do these checks below when going through the `modified_ratoml_files`, gets rid of these two extra variables here. Can also put the id into the `_text` vars instead then, allowing us to omit the unwraps below